### ISSUE TITLE
CategoryService: add - search - delete - ...

### ISSUE THREAD

**Issue by kewde** [particl/particl-desktop#863]
Implement the category API

- [ ] Create interfaces for commands & responses

```
>> help category
list   -  List all the item categories. 
get (<categoryId>|<categoryKey>)  -  Command for getting an item category associated with category Id or key 
    <categoryId>                  - Numeric - The ID belonging to the category we 
                                     want to retrive. 
    <categoryKey>                 - String - The key that identifies the category 
                                     we want to retrieve. 
add <categoryName> <description> (<parentItemCategoryId>|<parentItemCategoryKey>)  -  Command for adding an item category. 
    <categoryName>                - String - The name of the category to create. 
    <description>                 - String - A description of the category to 
                                     create. 
    <parentItemCategoryId>        - Numeric - The ID of the parent category of the 
                                     category we're creating. 
    <parentItemCategoryKey>       - String - The identifying key of the parent 
                                     category of the category we're creating. 
update <categoryId> <categoryName> <description> [<parentItemCategoryId>]  -  Update the details of an item category given by categoryId. 
    <categoryId>                  - Numeric - The ID of the category we want to 
                                     update. 
    <categoryName>                - String - The new name of the category we want 
                                     to update. 
    <description>                 - String - The new description of the category 
                                     we want to update. 
    <parentItemCategoryId>        - [optional] Numeric - The ID that identifies the 
                                     new parent category of the category we want to 
                                     update; default is the root category. 
remove <categoryId>  -  Remove and destroy an item category via categoryId. 
    <categoryId>                  - Numeric - The ID belonging to the category we 
                                     want to destroy. 
search [<searchString>]  -  Command for getting an item categories search by particular search string 
    <searchString>                - [optional] String - A search string for finding 
                                     categories by name.
```

**Comment by anandsinghparihar** [comment]
Hi @pciavald  using `depcheck` I can see following Unused dependencies:

**Unused dependencies**
*  angular/http
*  angular/platform-server
* font-awesome
* lodash.get
* lodash.isempty
* lodash.values
* open-sans-all

**Unused devDependencies**
* angular/compiler-cli
* types/jasmine
* types/node
* codelyzer
* karma-cli
* tslint
* tslint-microsoft-contrib
* typescript

**Missing dependencies**
* mkdirp
* app

I can see all packaged are required so could you please suggest me which `Unused dependencies` I need to remove.

**Comment by vikas-cis** [comment]
hey @kewde can you summarize what needs to complete in this issue as I can see category service is already created 

Thanks

**Comment by kewde** [comment]
We'll skip this for now and stick to the default categories.
